plugins/seen/plugin.py

```diff
@@ -0,0 +1,95 @@
+from datetime import datetime
+
+from cardinal.decorators import command, help, regex, event
+
+
+class SeenPlugin(object):
+    def __init__(self, cardinal):
+        self.users_seen = dict()
+        self.users_action = dict()
```

Comment:
Can we use a single dict for this, just to avoid any subtle bugs where a nick may exist in one dict, but not the other? I can imagine it keyed by nick, with a tuple or dict as its values. The inner tuple/dict would contain the last seen time as well as the action.